tests/test_middleware.py

```diff
@@ -46,3 +48,13 @@ def test_routing_middleware(middleware_app):
     assert (
         response.headers.get("operation_id") == "fakeapi.hello.post_greeting"
     ), response.status_code
+
+
+async def test_lifecycle():
```

Comment:
This test fails on python 3.7 because `AsyncMock` is only availabe from Python 3.8.
We skip other tests relying on AsyncMock for Python 3.7